packages/flipper-plugin-react-native-performance/android/FlipperReactPerformancePlugin.java

```diff
@@ -0,0 +1,148 @@
+    private long contentAppeared = 0;
+    private long bundleSize;
+    private final ReactInstanceManager reactInstanceManager;
+
+    public FlipperReactPerformancePlugin(ReactInstanceManager reactInstanceManager) {
+        this.reactInstanceManager = reactInstanceManager;
+
+        sessionStartTime = System.currentTimeMillis();
+
+        measureNativeStartupTime();
+
+        measureBundleSize();
+
+        addReactMarkers();
+    }
+
+    private void measureBundleSize() {
+        String jsBundleFile = reactInstanceManager.getDevSupportManager().getDownloadedJSBundleFile();
+        File file = new File(jsBundleFile);
+        bundleSize = file.length();
+    }
+
+    private void measureNativeStartupTime() {
+        try {
+            long processStartTime = FlipperReactPerformancePlugin.getStartTime(Process.myPid());
+            nativeStartupDuration = SystemClock.elapsedRealtime() - processStartTime;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addReactMarkers() {
+        ReactMarker.addListener(
+                (name, tag, instanceKey) -> {
+                    switch (name) {
+                        case DOWNLOAD_START:
+                            scriptDownloadStart = System.currentTimeMillis();
+                            break;
+                        case DOWNLOAD_END:
+                            scriptDownloadEnd = System.currentTimeMillis();
+                            break;
+                        case RUN_JS_BUNDLE_START:
+                            scriptExecutionStart = System.currentTimeMillis();
+                            break;
+                        case RUN_JS_BUNDLE_END:
+                            scriptExecutionEnd = System.currentTimeMillis();
+                            break;
+                        case CONTENT_APPEARED:
+                            contentAppeared = System.currentTimeMillis();
+                            sendMeasurements();
+                            break;
+                        case RELOAD:
+                            sessionStartTime = System.currentTimeMillis();
+                            break;
+                    }
+                });
+    }
+
+    private static long getStartTime(final int pid) throws IOException {
+        final String path = "/proc/" + pid + "/stat";
+        final BufferedReader reader = new BufferedReader(new FileReader(path));
+        final String stat;
+        try {
+            stat = reader.readLine();
+        } finally {
+            reader.close();
+        }
+        final String field2End = ") ";
+        final String fieldSep = " ";
+        final int fieldStartTime = 20;
+        final int msInSec = 1000;
+        try {
+            final String[] fields = stat.substring(stat.lastIndexOf(field2End)).split(fieldSep);
+            final long t = Long.parseLong(fields[fieldStartTime]);
+            int tckName;
+            try {
```

Comment:
If I'm correct this API is available on Android SDK >= 21, and the min supported SDK on the upcoming version of React Native 0.64 (currently in RC) is 21, so I think we can drop the meta programming.

Reply:
That is right. Okay, i changed it to not use meta programming but added a check for any case.